bulk-downloader.md

```diff
@@ -0,0 +1,79 @@
+# dotstorage bulk downloader
+
+## Context
+
+> Provide a bulk downloader for nft.storage https://github.com/w3s-project/project-tracking/issues/77
+
+### Timeline
+
+Last day of June
+
+### Suggested end solutions
+
+There are different options how the solution could look like:
+
+1. users just receive a download link
+2. users can still login into the console and request bulk download
+3. or any other idea
+
+## Built tools partially related
+
+### w3 migration tool
+
+- https://github.com/web3-storage/migrate-to-w3up
+- Enables [web3.storage](http://web3.storage) users from [old.web3.storage](http://old.web3.storage) to migrate their uploads to a new account in w3up
+    - Given [web3.storage](http://web3.storage) already hosts users’ data, this tool can follow an efficient solution of not needing to download and re-upload all the data to the new service.
+    - In fact, service will figure out that it already stores the data out of the box when a user asks to store some CID. As a result, this tool simply goes through all the uploads in a [old.web3.storage](http://old.web3.storage) account and performs `store/add` invocations to make content associated with the new space for the user in w3up. Given content will be already stored, no bytes need to be wired.
+
+## Risks and limitations with a bulk downloader as a API
+
+The suggested end solutions above assume that an API is provided for this bulk download, i.e. all the work is done within service backend. It is important to consider the risks and limitations of such solution architecture.
+
+This project targets a completely different goal than w3up migration tool. Here the goal would be to actually download the entire content previously uploaded by an account. This solution is technically challenging, as well as expensive for the service operator. In addition, it comes at a cost of losing these users. 
+
+Considering the initial plan to provide a bulk download URL, there are a set of drawbacks and concerns that should be taken into account, specially when users have hundreds of GBs on their accounts:
+
+- Not fault tolerant to errors
+    - network errors no recoverable
+    - no more space on disk
+- Not verifiable
+    - we would not be served content addressed data that can be verified, but a compacted file with all the data
```

Comment:
Could offset, range, hash data be provided to allow content verification? Would there be a sufficient use-case for that?